src/entity/user/DeletedUser.ts

```diff
@@ -0,0 +1,10 @@
+import { CreateDateColumn, Entity, PrimaryColumn } from 'typeorm';
+
+@Entity()
+export class DeletedUser {
```

Comment:
Opted to create new entity which we can check on user creation, instead of adding `deleted` column to `User` entity which we would then need to check on all other places.

Reply:
I agree. Using soft delete on user table would have forced us to be creative when the same user tries to register with the same email after deleting a previous account for example